src/main/java/spireMapOverhaul/zones/gremlincamp/monsters/GremlinCook.java

```diff
@@ -0,0 +1,146 @@
+import spireMapOverhaul.util.Wiz;
+import spireMapOverhaul.zones.gremlincamp.PlayerPoisonPower;
+
+import java.util.ArrayList;
+
+import static spireMapOverhaul.SpireAnniversary6Mod.makeID;
+
+public class GremlinCook extends AbstractSMOMonster {
+    public static final String ID = makeID(GremlinCook.class.getSimpleName());
+    private static final MonsterStrings monsterStrings = CardCrawlGame.languagePack.getMonsterStrings(ID);
+    public static final String NAME = monsterStrings.NAME;
+    public static final String[] MOVES = monsterStrings.MOVES;
+
+    private static final Byte ALL_BUFF = 0, ALL_BULK = 1, POISON = 2, TACKLE = 3;
+
+    private int healAmt, strAmt, poisonAmt;
+    private boolean secondMove = false;
+
+    public GremlinCook() {
+        this(0, 0);
+    }
+
+    public GremlinCook(float x, float y) {
+        super(NAME, ID, 50, 0, 0, 120f, 210f, SpireAnniversary6Mod.makeMonsterPath("GremlinCamp/GremlinCook.png"), x, y);
+        setHp(calcAscensionTankiness(56), calcAscensionTankiness(62));
+        addMove(ALL_BUFF, Intent.BUFF);
+        addMove(ALL_BULK, Intent.BUFF);
+        addMove(POISON, Intent.STRONG_DEBUFF);
+        addMove(TACKLE, Intent.ATTACK_DEBUFF, calcAscensionDamage(10));
+
+        healAmt = calcAscensionSpecial(8);
+        strAmt = AbstractDungeon.ascensionLevel >= 18? 3: 2;
+        poisonAmt = calcAscensionSpecial(6);
+    }
+
+    @Override
+    protected void setUpMisc() {
+        super.setUpMisc();
+        this.type = EnemyType.NORMAL;
+    }
+
+    @Override
+    public void takeTurn() { //TODO: Add sounds
+        switch (nextMove) {
+            case 0:
+                useShakeAnimation(Settings.ACTION_DUR_MED);
+                Wiz.atb(new AllEnemyApplyPowerAction(this, strAmt, (m) -> new StrengthPower(m, strAmt)));
+                break;
+            case 1:
+                useShakeAnimation(Settings.ACTION_DUR_FAST);
+                Wiz.atb(new AbstractGameAction() {
+                    @Override
+                    public void update() {
+                        isDone = true;
+                        Wiz.forAllMonstersLiving(m -> {
+                            if(!GremlinCook.ID.equals(m.id)) {
+                                m.increaseMaxHp(healAmt, true);
+                                AbstractDungeon.effectList.add(new HealEffect(m.hb.cX, m.hb.cY, healAmt));
+                            }
+                        });
+                    }
+                });
+                break;
+            case 2:
+                useFastAttackAnimation();
+                Wiz.atb(new ApplyPowerAction(AbstractDungeon.player, this, new PlayerPoisonPower(AbstractDungeon.player, poisonAmt), poisonAmt, AbstractGameAction.AttackEffect.POISON));
+                break;
+            case 3:
+                DamageInfo info = new DamageInfo(this, moves.get(nextMove).baseDamage, DamageInfo.DamageType.NORMAL);
+                if (info.base > -1) {
```

Comment:
What's the purpose of this check? Don't we know what the base damage for this move is? I'm guessing this was just copied from somewhere and/or that it was part of some generic check that was copied over for a bunch of moves that weren't all attacks.

So I guess that makes sense, still confusing to read though. I'd want to get rid of it.